### PR TITLE
22w13a client

### PR DESCRIPTION
**Pull request by apple502j**

### CODE REVIEW

File: mappings/net/minecraft/client/option/SimpleOption.mapping [hunk 1]
```diff
@@ -1,4 +1,66 @@
...
+	COMMENT trigger with the new value.
+	COMMENT
+	COMMENT <h2 id="callbacks">Callbacks</h2>
+	COMMENT <p>This class itself cannot be extended by default; instead, the behavior of the option,
+	COMMENT such as validation or rendering of the {@link ClickableWidget} associated with the
+	COMMENT option, is customized by passing a {@link SimpleOption.Callbacks} instance. There are
+	COMMENT several existing callbacks that should cover most of the needs:
+	COMMENT
+	COMMENT <ul>
+	COMMENT <li>{@link SimpleOption.PotentialValuesBasedCallback}: The most simple cycling option.
```
Comment (enbrain) [inline mappings/net/minecraft/client/option/SimpleOption.mapping]:
```suggestion
	COMMENT <li>{@link SimpleOption.PotentialValuesBasedCallbacks}: The most simple cycling option.
```

File: mappings/net/minecraft/client/option/SimpleOption.mapping [hunk 2]
```diff
@@ -1,4 +1,66 @@
...
+	COMMENT with a minimum and maximum values. Values outside this range are considered invalid.</li>
+	COMMENT </ul>
+	COMMENT
+	COMMENT <p>See also several {@code ofBoolean} methods for options using boolean values.
+	COMMENT
+	COMMENT <h3 id="callbacks-modifiers">Modifiers</h3>
+	COMMENT <p>Slider option callbacks allow "modifiers" to be applied. The "slider progress value"
+	COMMENT is the value used to calculate the slider's progress and must be linear. The real value,
+	COMMENT simply called "value" in {@code withModifier} methods, is the value used in
+	COMMENT {#link getValue()} and passed to {@link #textGetter}.
```
Comment (enbrain) [inline mappings/net/minecraft/client/option/SimpleOption.mapping]:
```suggestion
	COMMENT {#link #getValue()} and passed to {@link #textGetter}.
```

File: mappings/net/minecraft/client/option/SimpleOption.mapping [hunk 3]
```diff
@@ -25,36 +96,82 @@ CLASS net/minecraft/class_7172 net/minecraft/client/option/SimpleOption
...
+		ARG 1 client
+	METHOD method_42720 enumValueText ()Lnet/minecraft/class_7172$class_7303;
+		COMMENT {@return the getter for the {@code valueTextGetter} parameter of the constructor
+		COMMENT to use the value's text as the option's rendered text}
+		COMMENT
+		COMMENT @apiNote This requires the value to be an enum extending {@link TranslatableOption}.
 	CLASS class_7173 PotentialValuesBasedCallbacks
+		COMMENT A set of callbacks for a cycling option.
+		COMMENT
+		COMMENT @see <a href="#callbacks">Callbacks</a>
```
Comment (enbrain) [inline mappings/net/minecraft/client/option/SimpleOption.mapping]: Isn't this link invalid since javadoc generates an html page for each inner class?

**Review by enbrain — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/option/SimpleOption.mapping (hunk 3 above)
Comment (apple502j) [reply mappings/net/minecraft/client/option/SimpleOption.mapping]: Hmm, how do we link to that then?

Comment (enbrain) [reply mappings/net/minecraft/client/option/SimpleOption.mapping]: Hmm, I can think of two ways.

```suggestion
		COMMENT @see <a href="SimpleOption.html#callbacks">Callbacks</a>
```

You can include the filename to specify the class. Intellij IDEA is smart enough to render this link properly. Eclipse and VSCode don't support the `<a>` tag anyway.

Alternatively, you could move docs to the `Callbacks` class and link to it instead.

**Pull request merged**